espresso/scripts/prepare-allocs.sh

```diff
@@ -104,6 +107,131 @@ BATCH_AUTHENTICATOR_OWNER_ADDRESS="${BATCH_AUTHENTICATOR_OWNER_ADDRESS}" op-depl
                   --private-key="${OPERATOR_PRIVATE_KEY}"
 
+# =====================================================================
+# Deploy op-succinct contracts (OPSuccinctFaultDisputeGame, AccessManager, SP1MockVerifier)
+# =====================================================================
+echo "Deploying op-succinct contracts..."
+
+# Configuration for op-succinct
+GAME_TYPE=42  # Succinct game type
+MAX_CHALLENGE_DURATION=300  # 5 minutes for devnet
+MAX_PROVE_DURATION=1800     # 30 minutes for devnet
+FALLBACK_TIMEOUT_FP_SECS=3600  # 1 hour for devnet
+INITIAL_BOND_WEI=1000000000000000  # 0.001 ETH
+CHALLENGER_BOND_WEI=1000000000000000  # 0.001 ETH
+DISPUTE_GAME_FINALITY_DELAY_SECONDS=6  # Fast for devnet
+
+# Read existing contract addresses from state.json
+DISPUTE_GAME_FACTORY=$(jq -r '.opChainDeployments[0].DisputeGameFactoryProxy' "${DEPLOYER_DIR}/state.json")
+OPTIMISM_PORTAL=$(jq -r '.opChainDeployments[0].OptimismPortalProxy' "${DEPLOYER_DIR}/state.json")
+ANCHOR_STATE_REGISTRY=$(jq -r '.opChainDeployments[0].AnchorStateRegistryProxy' "${DEPLOYER_DIR}/state.json")
+
+echo "Existing contract addresses:"
+echo "  DisputeGameFactory: ${DISPUTE_GAME_FACTORY}"
+echo "  OptimismPortal: ${OPTIMISM_PORTAL}"
+echo "  AnchorStateRegistry: ${ANCHOR_STATE_REGISTRY}"
+
+# Get the starting output root from the anchor state registry
+# The AnchorStateRegistry was initialized by op-deployer with a starting anchor root
+STARTING_ROOT=$(cast call "${ANCHOR_STATE_REGISTRY}" "getAnchorRoot()(bytes32,uint256)" --rpc-url "${ANVIL_URL}" | head -1)
+STARTING_L2_BLOCK=$(cast call "${ANCHOR_STATE_REGISTRY}" "getAnchorRoot()(bytes32,uint256)" --rpc-url "${ANVIL_URL}" | tail -1)
+
+echo "Starting anchor state:"
+echo "  Root: ${STARTING_ROOT}"
+echo "  L2 Block: ${STARTING_L2_BLOCK}"
+
+# Always create succinct.env with contract addresses (even if op-succinct repo not found)
+# Note: op-succinct uses FACTORY_ADDRESS not DISPUTE_GAME_FACTORY_ADDRESS
+SUCCINCT_ENV_FILE="${DEPLOYER_DIR}/succinct.env"
+cat > "${SUCCINCT_ENV_FILE}" << EOF
+FACTORY_ADDRESS=${DISPUTE_GAME_FACTORY}
+DISPUTE_GAME_FACTORY_ADDRESS=${DISPUTE_GAME_FACTORY}
+OPTIMISM_PORTAL2_ADDRESS=${OPTIMISM_PORTAL}
+ANCHOR_STATE_REGISTRY_ADDRESS=${ANCHOR_STATE_REGISTRY}
+EOF
+echo "Created succinct env file at ${SUCCINCT_ENV_FILE}"
+
+# Deploy op-succinct contracts using local contracts-bedrock deployment script
+
+echo "Deploying op-succinct contracts from local packages/contracts-bedrock..."
+
+# Export environment variables consumed by DeployOPSuccinctFDG.s.sol
+export FACTORY_ADDRESS="${DISPUTE_GAME_FACTORY}"
+export ANCHOR_STATE_REGISTRY_ADDRESS="${ANCHOR_STATE_REGISTRY}"
+export GAME_TYPE="${GAME_TYPE}"
+export INITIAL_BOND_WEI="${INITIAL_BOND_WEI}"
+export CHALLENGER_BOND_WEI="${CHALLENGER_BOND_WEI}"
+export MAX_CHALLENGE_DURATION="${MAX_CHALLENGE_DURATION}"
+export MAX_PROVE_DURATION="${MAX_PROVE_DURATION}"
+export USE_SP1_MOCK_VERIFIER="true"
+export PERMISSIONLESS_MODE="true"
```

Comment:
According to the [OP Succinct doc](https://succinctlabs.github.io/op-succinct/fault_proofs/deploy.html?ref=blog.succinct.xyz#required-environment-variables), `PROPOSER_ADDRESS` will be "Ignored if PERMISSIONLESS_MODE is true". Should we remove the use of `PROPOSER_ADDRESS` when running the Succinct mode?